src/devices/Mcp3428/Mcp3428.cs

```diff
@@ -0,0 +1,409 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Device.I2c;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Mcp3428
+{
+    public partial class Mcp3428 : IDisposable
+    {
+        public Mcp3428(I2cDevice i2CDevice) : this(i2CDevice, 4)
+        {
+        }
+
+        public Mcp3428(I2cDevice i2CDevice, ModeEnum mode = ModeEnum.Continuous,
+            ResolutionEnum resolution = ResolutionEnum.Bit12, GainEnum pgaGain = GainEnum.X1) : this(i2CDevice)
+        {
+            SetConfig(0, mode: mode, resolution: resolution, pgaGain: pgaGain);
+        }
+
+        /// <summary>
+        /// Protected constructor to easily generate MCP3426/7 devices whose only difference is channel count and I2C address
+        /// </summary>
+        /// <param name="i2CDevice">The i2 c device.</param>
+        /// <param name="channels">The channels.</param>
+        /// <autogeneratedoc />
+        protected Mcp3428(I2cDevice i2CDevice, int channels)
+        {
+            _i2cDevice = i2CDevice;
+            ChannelCount = channels;
+            ReadValue(); // Don't like this in constructor, makes sure props are valid
```

Comment:
not sure I understand the comment, what happens if you remove this?

Reply:
I don't like initiating communication in the constructor.
This method ensures that if we are not after reset and previous changes were made to the device configuration then these changes are reflected in the created object.
The class is designed to always mirror hardware configuration state so this read call is required to initialize an instance.